### PR TITLE
[bitnami/etcd] Add support for image digest apart from tag

### PR DESCRIPTION
**Pull request by carrodher**
### Description of the change

From now on it will be possible to set an `image.digest` instead of `image.tag` to pull the container image. Please note it is needed to release a bitnami/common version with those changes (see https://github.com/bitnami/charts/pull/11797).

Once applied the changes, this is the result when setting the `image.digest` parameter in the _values.yaml_:
```yaml
## Bitnami etcd image version
## ref: https://hub.docker.com/r/bitnami/etcd/tags/
## @param image.registry etcd image registry
## @param image.repository etcd image name
## @param image.tag etcd image tag
## @param image.digest etcd image digest in the way sha256:aa.... Please note this parameter, if set, will override the tag
##
image:    
  registry: docker.io
  repository: bitnami/etcd
  tag: 3.5.4-debian-11-r22
  digest: sha256:507bc997779af5f0419ad917167ba1c9a2ceb936f2c1e82fb0f687e6c1296897
```
```console
$ helm template . -s templates/statefulset.yaml | grep 'image:'
          image: docker.io/bitnami/etcd@sha256:507bc997779af5f0419ad917167ba1c9a2ceb936f2c1e82fb0f687e6c1296897
```
In the same way, when the `image.digest` is empty (default value), this is the result:
```yaml
## Bitnami etcd image version
## ref: https://hub.docker.com/r/bitnami/etcd/tags/
## @param image.registry etcd image registry
## @param image.repository etcd image name
## @param image.tag etcd image tag
## @param image.digest etcd image digest in the way sha256:aa.... Please note this parameter, if set, will override the tag
##
image:    
  registry: docker.io
  repository: bitnami/etcd
  tag: 3.5.4-debian-11-r22
  digest: ""
```
```console
$ helm template . -s templates/statefulset.yaml | grep 'image:'
          image: docker.io/bitnami/etcd:3.5.4-debian-11-r22
```

### CODE REVIEW

File: bitnami/etcd/values.yaml [hunk 1]
```diff
@@ -65,19 +65,20 @@ diagnosticMode:
 ## @param image.registry etcd image registry
 ## @param image.repository etcd image name
 ## @param image.tag etcd image tag
-## @param image.pullPolicy etcd image pull policy
-## @param image.pullSecrets [array] etcd image pull secrets
-## @param image.debug Enable image debug mode
+## @param image.digest etcd image digest in the way sha256:aa...
```
Comment (rafariossaa) [inline bitnami/etcd/values.yaml:68]: I would add a note about `digest` having precedence on `tag`

Comment (carrodher) [inline bitnami/etcd/values.yaml:68]:
```suggestion
## @param image.digest etcd image digest in the way sha256:aa.... Please note this parameter, if set, will override the tag
```

File: bitnami/etcd/values.yaml [hunk 2]
```diff
@@ -589,14 +591,17 @@ volumePermissions:
   ## @param volumePermissions.image.registry Init container volume-permissions image registry
   ## @param volumePermissions.image.repository Init container volume-permissions image name
   ## @param volumePermissions.image.tag Init container volume-permissions image tag
-  ## @param volumePermissions.image.pullPolicy Init container volume-permissions image pull policy
-  ## @param volumePermissions.image.pullSecrets [array] Specify docker-registry secret names as an array
+  ## @param volumePermissions.image.digest Init container volume-permissions image digest in the way sha256:aa...
```
Comment (carrodher) [inline bitnami/etcd/values.yaml:594]:
```suggestion
  ## @param volumePermissions.image.digest Init container volume-permissions image digest in the way sha256:aa.... Please note this parameter, if set, will override the tag
```

**Review by rafariossaa — DISMISSED** [review]
LGTM

**Review by rafariossaa — APPROVED** [review]
LGTM

**Pull request merged**